libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -110,29 +110,31 @@ void AP_FETtecOneWire::update()
     TelemetryData t {};
     int16_t centi_erpm;
     uint16_t tx_err_count;
+    uint8_t tlm_ok = 0; //decode_single_esc_telemetry returns 1 if telemetry is ok, 0 if its waiting and 2 if there is a crc mismatch.
+    uint8_t tlm_from_id = 0;
     if (_requested_telemetry_from_esc) {
-        decode_single_esc_telemetry(t, centi_erpm, tx_err_count);
+        tlm_ok = decode_single_esc_telemetry(t, centi_erpm, tx_err_count, tlm_from_id);
     }
-    if (_requested_telemetry_from_esc == nr_escs || _active_esc_ids[_requested_telemetry_from_esc-1] == 0) {
-        _requested_telemetry_from_esc = 1; // restart from ESC0
+    if (_requested_telemetry_from_esc == _found_escs_count) { //if found esc number is reached restart request counter
+        _requested_telemetry_from_esc = 1; // restart from ESC 1 
     } else {
         _requested_telemetry_from_esc++;
     }
 #endif
-
-    escs_set_values(motor_pwm, nr_escs, _requested_telemetry_from_esc);
+   
+    escs_set_values(motor_pwm, _found_escs_count, _requested_telemetry_from_esc);

```

Comment:
why are you using _found_escs_count instead of nr_escs?

Another question: can this parameter be removed? it has no effect inside that function. I think we should validate if nr_escs == _found_escs_count at some point.